src/crab/linear_expression.hpp

```diff
@@ -101,8 +101,10 @@ class linear_expression_t final {
     [[nodiscard]] linear_expression_t operator-(const variable_t& variable) const {
         variable_terms_t variable_terms = _variable_terms;
         number_t new_coefficient = coefficient_of(variable) - 1;
-        if (new_coefficient != 0) {
+        if (!new_coefficient.fits_sint() || ((int)new_coefficient) != 0) {
```

Comment:
Can you explain this change? I don't think I understand it.
Also, what is the difference between this case and the operator implemented below (subtracting `linear_expression_t`)? Subtracting a variable is a special case, so I would expect similar handling below.

Reply:
when the new_coefficient was 0, the old line was still true for some reason, whereas the new line now correctly is false.  You're probably right that the one below also needs a fix.  I will try to find a test case that hits that path.

Reply:
fixed

Reply:
I think the same issue exists in `operator+()` and (maybe) `operator*`. If I understand it correctly, it as a normalization issue, so a less repetitive solution would be to add `normalize(variable_terms)` before constructing the result - or probably better, in the constructor, since the class is essentially immutable. What do you think?

Reply:
Yes you're right about it existing in operator+ as well now that I look there, and I suppose operator* if constant is a 0

Reply:
fixed

Reply:
I think the other constructor needs a conditional too:
```
linear_expression_t(const number_t& coefficient, const variable_t& variable) { _variable_terms[variable] = coefficient; }
```

Reply:
done